LevelUp.Pos.ProposedOrders/ProposedOrderCalculator.cs

```diff
@@ -88,23 +94,43 @@ int customerPaymentAmount
             int appliedDiscountAmount
         )
         {
-            AdjustedCheckValues values = CalculateCreateProposedOrderValues(
-                totalOutstandingAmount,
+            int totalOutstandingAmountWithDiscount = totalOutstandingAmount + Math.Abs(appliedDiscountAmount);
```

Comment:
@devlaf, you had an alternate suggestion for a way to approach these calculations by adjusting the requested spend relative to the discount applied.  Would you outline that here for me?

Reply:
```C#
 public static AdjustedCheckValues CalculateCompleteOrderValues(
            int totalOutstandingAmount,
            int totalTaxAmount,
            int totalExemptionAmount,
            int customerPaymentAmount,
            int appliedDiscountAmount
        )
        {
            int remainingSpendAmount = Math.Max(0, Math.Min(customerPaymentAmount - appliedDiscountAmount, totalOutstandingAmount));

            int adjustedTaxAmount = CalculateAdjustedTaxAmount(
                totalOutstandingAmount, 
                totalTaxAmount,
                remainingSpendAmount);

            int adjustedExemptionAmount = CalculateAdjustedExemptionAmount(
                totalOutstandingAmount,
                totalTaxAmount,
                totalExemptionAmount, 
                remainingSpendAmount);

            int adjustedSpendAmount = remainingSpendAmount + appliedDiscountAmount;

            return new AdjustedCheckValues(adjustedSpendAmount, adjustedTaxAmount, adjustedExemptionAmount);
        }
```